### ISSUE TITLE
use /stretchr/testify package instead of own assert functions for testing

### ISSUE THREAD

**Issue by BigJk** [kasvith/kache#20]
**Description**

I think it's not needed to reinvent the wheel when there is already a great and feature packed assert package that has everything that could be needed to test the codebase. Please checkout [testify](https://github.com/stretchr/testify) [(godoc)](https://godoc.org/github.com/stretchr/testify/assert).

**Steps to reproduce the behavior:**

**Expected behavior**

**OS and Other informations**

**Additional details**

**Comment by kasvith** [comment]
But does it support things like **AssertStringSliceEqual** or **ContainsElements** like testing scenarios ?

We need it to assert some types

**Comment by BigJk** [comment]
- **AssertStringSliceEqual**
You are using ``reflect.DeepEqual`` [(ref)](https://github.com/kasvith/kache/blob/master/pkg/testsuite/assert.go#L40) to check the equality. In testify the ``Equal`` [(ref)](https://github.com/stretchr/testify/blob/master/assert/assertions.go#L337) method calls the ``ObjectsAreEqual`` [(ref)](https://github.com/stretchr/testify/blob/26d6fa081fa47d3249e09a12cf434fe1adc5ef96/assert/assertions.go#L63) method which itself uses ``reflect.DeepEqual``. So in the end it seems like the same. 

- **ContainsElements**
Testify has a ``Contains`` function suitable to check if a element is in a slice or array. You are right that there is no single function to check if multiple elements are in a slice or array but it's easy to write one little helper function or just use a little loop where needed.

**Example:** 

```go
keys := hm.Keys()
testsuite.ContainsElements(t, elements, keys)
```

will be

```go
keys := hm.Keys()
for i := range keys {
	assert.Contains(t, elements, keys[i])
}
```

**Comment by kasvith** [comment]
Yes we can go with this.

**Comment by Chyroc** [comment]
this is closed by my commit message `fix #20`, and this should be reopen
